### PR TITLE
Expose Uphold Status and ID

### PR DESCRIPTION
**Pull request by corymcdonald**
Changed the following

* We can actually search by a UUID now
* We can search for identical Uphold accounts
* Rewrote the `payout_report_publisher_includer_test` to handle more permutations of statuses
* Sorting by Last Signed In will show the null values last so we can see who was the most recent signed in user

### CODE REVIEW

File: app/views/admin/publishers/show.html.slim [hunk 1]
```diff
@@ -18,11 +18,19 @@
...
-          .db-field = "Uphold verified:"
+          .db-field = "Uphold once connected:"
           .db-value = @publisher.uphold_verified?
         .db-info-row
           .db-field = "Uphold authorized:"
           .db-value = @publisher.wallet.authorized?
+        - if @publisher.wallet.status.present?
+          .db-info-row
+            .db-field = "Uphold status:"
+            .db-value = @publisher.wallet.status
```
Comment (yachtcaptain23) [inline app/views/admin/publishers/show.html.slim]: IIRC this isn't necessarily the status we would want. I was getting `re-authorize` for my account, so I believe this comes from Eyeshade/Ledger and not from Uphold.

File: config/locales/en.yml [hunk 2]
```diff
@@ -293,6 +293,15 @@ en:
         </p>
         <p>Your earnings will be paid out in the next payment cycle after you verify your identity with Uphold. In the meantime, your earnings will still be held safely in your account. We’re sorry for the inconvenience and delays this may cause. Thank you for being a Brave Creator and bearing with us as we improve the integrity of the payout system. Learn more about this new requirement <a href="https://support.uphold.com/hc/en-us/articles/202887565-What-are-the-benefits-of-becoming-a-verified-member-of-Uphold-">here</a>.</p>
         <p>Thank you,</p>
+    uphold_member_restricted:
+      subject: Please check Uphold Accoun
```
Comment (yachtcaptain23) [inline config/locales/en.yml]: Spelling

File: config/locales/en.yml [hunk 3]
```diff
@@ -293,6 +293,15 @@ en:
...
+      body_html: |
+        <p>Thank you for being a Brave creator.</p>
+        <p>
+          We've noticed that your Uphold account is not able to receive funds. Plus check your Uphold account status and ensure that there are no actionable items and that you are within the transaction limit guidelines.
```
Comment (yachtcaptain23) [inline config/locales/en.yml]: Spelling

File: lib/eyeshade/wallet.rb [hunk 4]
```diff
@@ -25,6 +27,8 @@ def initialize(wallet_json:, channel_json:)
       @possible_currencies = details_json["possibleCurrencies"] || []
       @address = details_json["address"] || ""
       @is_member = details_json["isMember"] || false
+      @status  = details_json["status"]
+      @uphold_id = details_json["id"]
```
Comment (yachtcaptain23) [inline lib/eyeshade/wallet.rb]: Do we know if it's `id` or `address`? @evq would you verify this?

Comment (evq) [reply lib/eyeshade/wallet.rb]: it's `id`. `address` is the card address of the first card that matches the user's preferred currency

File: config/locales/en.yml (hunk 3 above)
Comment (corymcdonald) [reply config/locales/en.yml]: https://github.com/brave-intl/publishers/pull/1538/commits/e3cadf7bf81d81eb370b561dacc2d28ea59069b0

File: config/locales/en.yml (hunk 2 above)
Comment (corymcdonald) [reply config/locales/en.yml]: https://github.com/brave-intl/publishers/pull/1538/commits/e3cadf7bf81d81eb370b561dacc2d28ea59069b0

File: app/controllers/admin/publishers_controller.rb [hunk 5]
```diff
@@ -92,4 +94,9 @@ def update_params
   def sortable_columns
     [:last_sign_in_at, :created_at, Publisher::VERIFIED_CHANNEL_COUNT]
   end
+
+  def is_a_uuid?(uuid)
+    uuid_regex = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/
```
Comment (nvonpentz) [inline app/controllers/admin/publishers_controller.rb]: if you didn't write this, we should probably credit where it came from in a comment

Comment (corymcdonald) [reply app/controllers/admin/publishers_controller.rb]: Ah good point - added the reference to the original location
https://github.com/brave-intl/publishers/pull/1538/commits/7deafa87df6a7870871f1f2ec588999d5c85e6b9

**Pull request merged**